src/extensions/replay/config.ts

```diff
@@ -79,19 +79,29 @@ const ignorePostHogPaths = (data: CapturedNetworkRequest): CapturedNetworkReques
     return data
 }
 
+function estimateBytes(payload: string): number {
+    return new Blob([payload]).size
+}
+
 function redactPayload(
-    payload: Body,
+    payload: string | null | undefined,
     headers: Record<string, any> | undefined,
     limit: number,
     description: string
-): Body {
-    const requestContentLength = headers?.['content-length']
-    if (_isUndefined(requestContentLength)) {
-        return `[SessionReplay] no content-length header for ${description}, cannot determine payload size`
+): string | null | undefined {
+    if (_isUndefined(payload) || _isNull(payload)) {
+        return payload
+    }
+
+    let requestContentLength: string | number = headers?.['content-length'] || estimateBytes(payload)
+    if (_isString(requestContentLength)) {
```

Comment:
turns out we rarely/never have the content-length header for requests

so we have to have a way to estimate body size when there is no header

changes elsewhere in this PR meant we know this is always a string now so that's much simpler than I was worrying

Reply:
That is odd and might be a problem, see this [issue](https://github.com/PostHog/posthog-android/pull/86).
Some providers require the `content-length` otherwise the provider will return 411 but we cannot control the user's requests, maybe something to add on the [docs](https://posthog.com/docs/session-replay/network-recording)?

Reply:
Ah, not that the request won't have it but `fetch` doesn't expose computed request headers to code here. So, we can't read content-length regardless of whether the browser adds it when sending.

I tried not reading the headers until after the request had been made and couldn't read request headers that were visible in the network panel of the browser

Reply:
I see, gotcha.